analyses/count-contributions/02-format-contributions.Rmd

```diff
@@ -0,0 +1,182 @@
+---
+title: "Format analyses directories contributions"
+output: html_notebook
+author: JN Taroni for ALSF CCDL
+date: 2022
+---
+
+## Set up
+
+```{r}
+library(tidyverse)
+```
+
+In the first script, we created TXT files that contain the authors of contributions to each module in `analyses/`.
+Those files live in a directory called `scratch/count-contributions/` (relative to the root of the repo).
+We're going to save the wrangled results to a `results/` directory
+
+```{r}
+components_dir <- file.path("..", "..", "scratch", "count-contributions")
+results_dir <- "results"
+dir.create(results_dir, showWarnings = FALSE)
+```
+
+We're going save the file that maps contributors to modules _and_ a file that counts the number of modules someone has contributed to.
+
+```{r}
+module_contributions_file <- file.path(results_dir, 
+                                       "module_contributors.tsv")
+count_modules_file <- file.path(results_dir,
+                                "module_contribution_counts.tsv")
+total_countributions_file <- file.path(results_dir,
+                                       "total_contributions_clean.tsv")
+```
+
+## Mapping different IDs to the same contributor
+
+Some folks contributed under two "names" (usually their GH handle and their name), or just their first name, or just their GH handle.
+So let's clean this up in the interest of readability in the final data frames.
+
+```{r}
+name_recoding <- c(
+  jashapiro = "Joshua Shapiro",
+  Stephanie = "Stephanie J. Spielman",
+  Bailey = "Bailey Farrow",
+  runjin326 = "Run Jin",
+  Mateusz = "Mateusz P. Koptyra",
+  tkoganti = "Teja Koganti",
+  NNoureen = "Nighat Noureen",
+  `Jo Lynne` = "Jo Lynne Rokita",
```

Comment:
Another alias...
```suggestion
  `Jo Lynne` = "Jo Lynne Rokita",
  jharenza = "Jo Lynne Rokita",
```

Reply:
Not currently necessary as far as I can tell, but sure I can add it. By logic, I'd add `jaclyn-taroni = "Jaclyn Taroni"` and so on, but that seems unmaintainable at best.